contest/contest_info.py

```diff
@@ -121,4 +126,28 @@ def can_ask(user,contest):
     return  user_is_contestant | user_is_owner_coowner
 
 def can_reply(user,contest):
-    return has_contest_ownership(user,contest)
+    return has_contest_ownership(user,contest) 
+
+def get_contest_or_404(contest_id):
+    try:
+        contest = Contest.objects.get(id = contest_id)
+        return contest
+    except Contest.DoesNotExist:
+        logger.warning('Contest: Can not register contest %s! Contest not found!' % contest_id)
+        raise Http404('Can not register contest %s! Contest not found!' % contest_id)
+
+def can_register(contest,user):
```

Comment:
You can use ```has_attend``` variable certainly. I suggested to to re-organize the ordering so that when we know a contest is not open register, we don't have to check other conditions.